Integrations/youkassa tests and scripts by Семиренко

AndVicSem112 opened the pull request:
Предыдущий PR https://github.com/carbonfay/DBCV/pull/171

## Описание изменений

Добавлены интеграции YooKassa: `youkassa_get_payments`, `youkassa_get_payment`, `youkassa_create_receipt` вместе с тестами и демонстрационным скриптом для локальной проверки без реальных credentials.

Изменения включают:
- Реализация классов интеграций в `backend/app/integrations/youkassa/`
  - `get_payments.py` — получение списка платежей через `yookassa.Payment.list`
  - `get_payment.py` — получение платежа по ID через `yookassa.Payment.find_one`
  - `create_receipt.py` — создание чека через `yookassa.Receipt.create`
- Тесты в `backend/app/tests/integrations/`
- Демонстрационный сценарий `scripts/test_yookassa_connection.py` — позволяет запускать интеграции локально с фиктивными данными из `backend/app/tests/fixtures/youkassa/`
- Регистрация интеграций в `backend/app/integrations/youkassa/__init__.py`

## Тип изменений

- [ ] Исправление бага
- [x] Новая функция
- [ ] Изменение документации
- [ ] Рефакторинг кода
- [x] Тесты
- [ ] Другое (опишите): демо-скрипт для QA

## Чеклист

- [x] Код соответствует стилю проекта
- [x] Я проверил свой код
- [x] Я добавил комментарии к коду, особенно в сложных местах
- [x] Я обновил документацию (включая примеры использования в metadata и демонстрационный скрипт)
- [x] Мои изменения не генерируют новых предупреждений
- [x] Я добавил тесты, которые подтверждают мои исправления
- [x] Новые и существующие тесты проходят локально (см. инструкцию ниже)

## Как протестировать

1. Убедитесь, что в вашем окружении установлены dev-зависимости (`pytest`, `pytest-asyncio`).
2. Запустите все интеграционные тесты:
   ```bash
   python -m pytest backend/app/tests/integrations -q
   ```
3. Для быстрой локальной проверки без установки pytest запустите демонстрацию:
   ```bash
   python scripts/test_yookassa_connection.py
   ```
   Она использует фикстуры в `backend/app/tests/fixtures/youkassa/` и выводит отчет `scripts/youkassa_test_results.json`.

## Дополнительная информация

- Фикстуры с примерными ответами (`payments_list.json`, `payment_detail.json`, `receipt_create.json`) находятся в отдельных ветках и PRs — это позволяет поэтапно ревьюить данные отдельно от кода.
- Live run results and analysis are available in `.github/pr_descriptions/youkassa-live-results.md` (contains requests/responses, explanations and recommended next steps).
- **Live run summary (added on branch update):** listing returned an empty payments list; standalone receipt attempts failed due to validation (see `scripts/youkassa_create_standalone_results.json`); scripts were updated to include stdlib HTTP implementation and Idempotence-Key handling for receipt creation attempts.
- Если хотите, могу автоматически создать PR тела из этих файлов или опубликовать результаты запуска в комментариях PR. 

# Live-run comment — feat/youkassa-tests-and-scripts

**Добавленные скрипты и результаты**

- `scripts/run_yookassa_live.py` — делает `GET /payments` и при найденном ID делает `GET /payments/{id}` (реализация на stdlib, чтобы не требовать установки сторонних пакетов).
- `scripts/create_standalone_receipts.py` — пытается создать 2 standalone-чека, добавляя `settlements`, `customer`, `Idempotence-Key` и логируя ответы.
- `scripts/test_yookassa_connection.py` — demo runner, использующий фикстуры (offline).

**Результаты live-прогона (summary):**

- `GET /payments?limit=20` → `items: []` (см. `scripts/youkassa_live_results.json`)
- Попытки создания standalone-чеки → валидация/ошибки (см. `scripts/youkassa_create_standalone_results.json`)
- Сценарии для воспроизведения и примеры запросов/ответов описаны в `.github/pr_descriptions/youkassa-live-results.md`

**Как воспроизвести (PowerShell):**

```powershell
$env:YOOKASSA_SHOP_ID='1236220'; $env:YOOKASSA_API_KEY='test_...'; python .\scripts\run_yookassa_live.py
$env:YOOKASSA_SHOP_ID='1236220'; $env:YOOKASSA_API_KEY='test_...'; python .\scripts\create_standalone_receipts.py
python scripts/test_yookassa_connection.py  # demo using fixtures
python -m pytest backend/app/tests/integrations -q  # unit tests
```

**Рекомендация:** основная рабочая траектория — найти реальные `payment_id` (через расширенный `GET /payments`) и затем создать чеки с `payment_id` и `send=false`. Скрипты и сохранённые результаты (`scripts/*.json`) помогут ревьюерам воспроизвести поведение.

**Файлы / доказательства:**

- `scripts/run_yookassa_live.py`
- `scripts/create_standalone_receipts.py`
- `scripts/youkassa_live_results.json`
- `scripts/youkassa_create_standalone_results.json`
- `scripts/youkassa_test_results.json`

AndVicSem112 committed:
feat(integration): add youkassa get_payment get_payments create_receipt integrations and tests

AndVicSem112 committed:
Add YooKassa payments_list fixture

AndVicSem112 committed:
Add YooKassa payment_detail fixture

AndVicSem112 committed:
Add YooKassa receipt_create fixture

AndVicSem112 committed:
Add YooKassa integrations, tests and demo script (fixtures are in separate branches)

AndVicSem112 committed:
Add PR description files for YooKassa fixture and integration branches

AndVicSem112 committed:
docs(tests-and-scripts): note live-run summary and script updates

Noted that listing returned empty payments, standalone receipt attempts failed, and scripts were updated to use stdlib HTTP + Idempotence-Key header; attached output files in scripts/*.json for reviewer.

AndVicSem112 committed:
scripts: add live-run scripts and output results

Added scripts/create_standalone_receipts.py and scripts/run_yookassa_live.py (stdlib HTTP implementation), and saved outputs: scripts/youkassa_create_standalone_results.json, scripts/youkassa_live_results.json, scripts/youkassa_test_results.json. This helps reviewers reproduce live runs and see exact request/response payloads.

AndVicSem112 committed:
docs: add PR comment file for feat/youkassa-tests-and-scripts

AndVicSem112 commented:
комментарии по вопросам из предыдущих коммитов вначале этого PR 
https://github.com/carbonfay/DBCV/pull/253

HadeSxGod1337 commented:
Спасибо за видео!
<img width="848" height="176" alt="image" src="https://github.com/user-attachments/assets/35bb48a8-5759-4347-9d4a-232ac953f84c" />
<img width="576" height="63" alt="image" src="https://github.com/user-attachments/assets/cdd40800-99db-460e-8693-f8b81931b1ab" />

Интеграция использует youkassa провайдер, но такого нет. Т.к. интеграция использует api_key, то провайдер надо выбрать other + api_key стратегию. Из-за чего собственно у вас не получилось протестировать интеграцию и показать ее работоспособность.  На видео вы верно выставили креды (other + api_key), вам лишь нужно поменять в коде (на скринах показал места где нужно исправить, но во всех интеграциях думаю найдете где).
После исправлений прошу показать результат работы интеграций. Если нужна помощь, можете обратится ко мне в личку в тг: @korshunov_hades.

Тестовые файлы расположены не в том месте. Необходимо размещать в backend\app\tests\integrations*your_integration**method_integration.py*

Также закоммичены лишние файлы, а именно в папке .github/...
<img width="396" height="125" alt="image" src="https://github.com/user-attachments/assets/53773924-9225-4e26-96bb-45bacb474f2b" />
и в папке scripts
<img width="409" height="99" alt="image" src="https://github.com/user-attachments/assets/27e5d1b2-c3b5-4087-8037-9f6c03a6fe4e" />
Эти файлы лишние.

Прошу исправить.

AndVicSem112 committed:
fix(youkassa): update credentials to support account_id/api_key variants, rename tests to match naming convention, remove unnecessary files

- Changed provider='youkassa' to provider='other' (as per project convention)
- Added support for multiple credential field names: account_id/shop_id, secret_key/api_key, oauth_token/token
- Renamed test files: test_yookassa_*.py -> youkassa_*_integration.py
- Removed unnecessary files: .github/pr_descriptions/comment_*.md, scripts/youkassa_*.json, scripts/run_yookassa_live.py, scripts/create_standalone_receipts.py

AndVicSem112 commented:
Всё исправил. Ошибка с авторизацией пропала, успешно достучался везде. 
Видео прикладываю, и скрин с логами из личного кабинета Youkassa 
Ссылка на видео https://skrinshoter.ru/vZFM5txTkJZ
Логи из кабинета 
<img width="1344" height="616" alt="image" src="https://github.com/user-attachments/assets/94727edd-4175-4fb5-a34f-664fe056421b" />

HadeSxGod1337 commented:
Есть недочеты:
В коммите присутствуют лишние файлы.

AndVicSem112 committed:
cleanup: remove PR descriptions and helper script per request

AndVicSem112 commented:
Почистил комиты, убрал все лишние файлы, остались только выбранные интеграции.
